### DISCUSSION TITLE
create an engine from pyodbc

### DISCUSSION THREAD

**Discussion by barmanroys** [sqlalchemy/sqlalchemy#8921] [Usage Questions]
Trying to write a pandas dataframe to a remote SQL database using the [to_sql][1] method. The pandas documentation says I have to supply an SQL Alchemy connectable as the ```con``` argument, but I am not clear how to form such a connectable object from the documentation. Basically, these are the parameters I have. 

```python
#!/usr/bin/env python3
# encoding: utf-8
import pandas as pd, pyodbc
server = '192.168.201.8' 
database = 'FORECAST' 
username = 'della' 
password = 'forecast948charcoal' 
# ENCRYPT defaults to yes starting in ODBC Driver 18. It's good to always specify ENCRYPT=yes on the client side to avoid MITM attacks.
# cnxn = pyodbc.connect('DRIVER={ODBC Driver 18 for SQL Server};SERVER='+server+';DATABASE='+database+';ENCRYPT=yes;UID='+username+';PWD='+ password)
cnxn = pyodbc.connect('DRIVER={ODBC Driver 18 for SQL Server};SERVER='+server+';TrustServerCertificate=yes;DATABASE='+database+';ENCRYPT=yes;UID='+username+';PWD='+ password)
df:pd.DataFrame=pd.read_csv(filepath_or_buffer='some_local_file.csv')
df.to_sql(name='test_table', con=cnxn) # pandas only supports SQLAlchemy connectable
```
So I realise probably the cnxn object is not useful for my purpose, but how do I get a sqlalchemy connectable using the parameters I have? 

**Platform details:**

Server: MSSQL 13.0.5026.0 on Windows 10 64 bit
Client Side:
- Python 3.8.10
- pandas 1.5.2
- pyodbc 4.0.35
- sqlalchemy 1.3.12
- ODBC Driver 18 for SQL Server
- Ubuntu 20.04

NOTE: 
I tested the connection credentials working on Azure data studio (from the same laptop where I am trying to run the python script). But I am not sure whether these same set of parameters are needed for SQL Alchemy. 

  [1]: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_sql.html

**Comment by CaselIT** [comment]
Hi,

You can start by looking at the docs https://docs.sqlalchemy.org/en/14/core/engines.html

**Comment by gordthompson** [comment]
https://stackoverflow.com/a/71083448/2144390
